src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs

```diff
@@ -147,15 +147,34 @@ private static bool IsNamespaceName(NameSyntax name)
             }
 
             // Use .Equals since we can't rely on object identity for constructed types.
-            if (symbol is ITypeSymbol typeSymbol)
+            SyntaxToken token;
+            switch (symbol)
             {
-                var classification = GetClassificationForType(typeSymbol);
-                if (classification != null)
-                {
-                    var token = name.GetNameToken();
-                    classifiedSpan = new ClassifiedSpan(token.Span, classification);
+                case ITypeSymbol typeSymbol:
+                    var classification = GetClassificationForType(typeSymbol);
+                    if (classification != null)
+                    {
+                        token = name.GetNameToken();
+                        classifiedSpan = new ClassifiedSpan(token.Span, classification);
+                        return true;
+                    }
+                    break;
+                case IFieldSymbol fieldSymbol:
+                    token = name.GetNameToken();
+                    classifiedSpan = new ClassifiedSpan(token.Span, ClassificationTypeNames.FieldName);
+                    return true;
+                case IMethodSymbol methodSymbol:
+                    token = name.GetNameToken();
+                    classifiedSpan = new ClassifiedSpan(token.Span, ClassificationTypeNames.MethodName);
```

Comment:
should we have different classification for method vs extension method?  I personally think yes.

Reply:
I like the idea to have extension methods in a different classification